docs/DeveloperGuide.md

```diff
@@ -1,34 +1,528 @@
+
+|Parameter<br />(Class Name)|Corresponds to<br />(Function of Class)|Referred to as<br />(Variable Name)|
+|:---:|:---:|:---:|
+|`ModuleLoader`| Class representing all modules offered by NUS | `allModules` |
+|`Person`| Class representing current user's information | `currentPerson`|
+|`Ui`| Class representing java's default scanner class | `in`|
+|`String` | Class representing the module code to be added | `moduleCode`|
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_state2.png" alt="State diagram for AddModuleCommand Step 2"/>
+</div>
+
+__Step 2__ : `execute()` is called from the instance of `AddModuleCommand`. It can throw `AcademicException` 
+or `IOException`. `FileHandler` and `Logger` classes from the _java API_ are instantiated to handle logging for the remainder of the `execute()` method. 
+
+__Step 3__ : `in` then reads in the next two lines of input, which is the user's input for the desired semester for the `moduleCode` and `moduleCode`'s grades.
+
+__Step 4__ : `validateInputs()` is called from `ModuleValidator` to validate the user entered data against `allModules`.
+
+__Step 5__ : `AddUtils` is called upon to return module credit for `moduleCode` by `getModuleCreditForModule()`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_state6.png" alt="State diagram for AddModuleCommand Step 6"/>
+</div>
+
+__Step 6__ :  `AddUtils` is called upon again to add the module's data to the user, by instantiating a new
+`PartialModule` and storing it in both `userModuleList` and `userModuleMap` via `Person`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_finalState.png" alt="Final state diagram for AddModuleCommand"/>
+</div>
+
+__Step 7__ : `FileHandler`, `Logger`, `PartialModule`, `ModuleValidator`, `AddUtils` and `AddModuleCommand` are terminated.
+
+The following sequence diagram shows how the `AddModuleCommand` works:
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_sequence.png" alt="Sequence diagram for AddModuleCommand"/>
+</div>
+
+The following activity diagram summarizes what happens when the user executes a `AddModuleCommand` :
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_activity.png" alt="Activity diagram for AddModuleCommand"/>
+</div>
+
+#### Design consideration
+
+The following options were considered when implementing commands:
+
+* Option 1 (Current Implementation): Implementing each command as a class by itself
+    * Pros: Increases modularity of code, higher overall code quality 
+    * Cons: More complicated to implement
+* Option 2: Implementing each command as a method in a class
+    * Pros: Easier to implement
+    * Cons: Class needs to be instantiated and increases coupling, reducing testability. This method also decreases SLAP.
```

Comment:
Maybe u can put this into table format so it's easier to read and do comparison between pros and cons